bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -241,3 +241,66 @@ def _build_udf_resources(resources):
         udf = {resource.udf_type: resource.value}
         udfs.append(udf)
     return udfs
+
+
+class ScalarQueryParameter(object):
+    """Named / positional query parameters for scalar values.
+
+    :type name: str or None
```

Comment:
In general, the `or None` isn't needed, instead we can say "(Optional) " in the prose (also `None` isn't a type)